### PR TITLE
Fix reading HUDI MOR Bootstrap tables in Hive connector

### PR DESCRIPTION
**Pull request by sanchitkashyap**
1. Removed  `isHudiRealtimeSplit(customSplitInfo)` condition in HiveUtil.java to support additional custom Hudi splits- BootstrapBaseFileSplit and RealtimeBootstrapBaseFileSplit. In general, any custom split if present should go through this conversion irrespective.
2. A new copy of JobConf is created in the case of BootstrapBaseFileSplit https://github.com/apache/hudi/blob/master/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java#L132, so we need to pass the actual config for copying to work correctly. More info #18911 .

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by sanchitkashyap** [comment]
@imjalpreet @pratyakshsharma can you pls review the change?

File: presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java
Comment (pratyakshsharma) [inline presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java:66]: Ideally you need another check to see if it is a `HoodieRealtimeBootstrapBaseFileSplit`.

I would rather remove the existing `isHudiRealtimeSplit` check as well since if fileSplit.getCustomSplitInfo() is not empty, it is one of HoodieRealtimeFileSplit/BootstrapBaseFileSplit/HoodieRealtimeBootstrapBaseFileSplit. This concept of CustomSplitConverter was introduced for hudi tables only. Hence checking the split type is redundant.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (pratyakshsharma) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java:319]: See my above comment. This method is probably not needed.

**Review by pratyakshsharma — CHANGES_REQUESTED** [review]
Thank you for raising this PR @sanchitkashyap 

Couple of comments, please check in your free time. Also good to get it reviewed from @bhasudha as well.

File: presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java
Comment (sanchitkashyap) [reply presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java]: Makes sense. I have removed the condition to check the split type.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java
Comment (sanchitkashyap) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: Removed.

**Comment by sanchitkashyap** [comment]
> Was wondering if it is feasible to write a small test case for this?
I have updated the tests in TestGenericHiveRecordCursorProvider.java. Pls take a look.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java
Comment (pratyakshsharma) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java:104]: Can we try to verify that the fileSplit behind this record cursor actually has BootstrapSplit? 

The reason I am thinking this is recordCursor was still present even before your changes.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java
Comment (pratyakshsharma) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java:97]: Lets verify the hudi split type here as well?

**Review by pratyakshsharma — CHANGES_REQUESTED** [review]
Thank you for adding the test cases @sanchitkashyap 

Hoepfully this is the last iteration and we can get it merged post you address the latest minor comments.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java
Comment (sanchitkashyap) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java:104]: Changes in commit 9dcdc46 are similar to the changes done here #19096 for MOR RT tables and these tests(added in 19096 only) just extend them for MOR bootstrap tables.
These test changes are not for the other commit that I have done in this PR.
I could not find any way to extract fileSplit from the record cursor.

File: presto-hive/src/main/java/com/facebook/presto/hive/util/HudiRealtimeBootstrapBaseFileSplitConverter.java
Comment (sanchitkashyap) [reply presto-hive/src/main/java/com/facebook/presto/hive/util/HudiRealtimeBootstrapBaseFileSplitConverter.java:39]: I think protected won't help in this case as this file is in a different package.

Comment (pratyakshsharma) [reply presto-hive/src/main/java/com/facebook/presto/hive/util/HudiRealtimeBootstrapBaseFileSplitConverter.java:39]: my bad.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java
Comment (pratyakshsharma) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java:104]: Thank you for confirming.

**Comment by pratyakshsharma** [comment]
@sanchitkashyap please squash the commits. 

@prestodb/committers This should be good to land. Please take a pass.

**Comment by sanchitkashyap** [comment]
> @sanchitkashyap please squash the commits.
I kept them as separate commits as they fix two different issues and are not exactly related.
Let me know if you still think that I should squash these commits.

**Comment by pratyakshsharma** [comment]
Both commits are needed to enable reading of bootstrap tables, and are meant to solve one bigger problem. I guess we can squash them. @sanchitkashyap

**Review by pratyakshsharma — APPROVED** [review]

**Comment by sanchitkashyap** [comment]
@zhenxiao This PR has already got one approval, can you pls review this as well?

**Comment by sanchitkashyap** [comment]
@zhenxiao Thanks for the review!
All the tests have passed. I think it is ready to be merged.

**Pull request merged**
